File: _2000/2501.py
import sys
sys.stdin = open('input.txt', 'r')
input = sys.stdin.readline

N, K = map(int, input().split())

l = [i for i in range(1, N+1) if N % i == 0]
if len(l) < K:
    print(0)
else:
    print(l[K-1])

File: _2000/2501.js
"strict mode";
const assert = require("node:assert/strict");
const stdin = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trimEnd()
  .split("\n");
const input = (() => {
  let line = 0;
  return () => stdin[line++];
})();

const get_divisor = (n) => {
  const divisors = [];
  const divisors_back = [];
  for (let i = 1; i <= Math.floor(n ** (1 / 2)); i++) {
    if (n % i == 0) {
      divisors.push(i);
      if (i == Math.floor(n / i)) continue;
      divisors_back.push(Math.floor(n / i));
    }
  }
  return divisors.concat(divisors_back.reverse());
};

const [N, K] = input().split(" ").map(Number);
const divisors = get_divisor(N);
console.log(divisors.length < K ? 0 : divisors[K - 1]);
